classes/announcement/Announcement.php

```diff
@@ -35,92 +35,76 @@ class Announcement extends \PKP\core\DataObject
     //
     /**
      * Get assoc ID for this announcement.
-     *
-     * @return int
      */
-    public function getAssocId()
+    public function getAssocId(): ?int
     {
         return $this->getData('assocId');
     }
 
     /**
      * Set assoc ID for this announcement.
-     *
-     * @param int $assocId
      */
-    public function setAssocId($assocId)
+    public function setAssocId(?int $assocId): void
     {
         $this->setData('assocId', $assocId);
     }
 
     /**
      * Get assoc type for this announcement.
-     *
-     * @return int
      */
-    public function getAssocType()
+    public function getAssocType(): ?int
     {
         return $this->getData('assocType');
     }
 
     /**
      * Set assoc type for this announcement.
-     *
-     * @param int $assocType
      */
-    public function setAssocType($assocType)
+    public function setAssocType(?int $assocType): void
     {
         $this->setData('assocType', $assocType);
     }
 
     /**
      * Get the announcement type of the announcement.
-     *
-     * @return int
      */
-    public function getTypeId()
+    public function getTypeId(): ?int
     {
         return $this->getData('typeId');
     }
 
     /**
      * Set the announcement type of the announcement.
-     *
-     * @param int $typeId
      */
-    public function setTypeId($typeId)
+    public function setTypeId(?int $typeId): void
     {
         $this->setData('typeId', $typeId);
     }
 
     /**
      * Get the announcement type name of the announcement.
-     *
-     * @return string|null
      */
-    public function getAnnouncementTypeName()
+    public function getAnnouncementTypeName(): ?string
     {
+        if (!$this->getData('typeId')) {
+            return null;
+        }
         $announcementTypeDao = DAORegistry::getDAO('AnnouncementTypeDAO'); /** @var AnnouncementTypeDAO $announcementTypeDao */
-        $announcementType = $announcementTypeDao->getById($this->getData('typeId'));
-        return $announcementType ? $announcementType->getLocalizedTypeName() : null;
+        return $announcementTypeDao->getById($this->getData('typeId'))->getLocalizedTypeName();
     }
 
     /**
      * Get localized announcement title
-     *
-     * @return string
      */
-    public function getLocalizedTitle()
+    public function getLocalizedTitle(): ?string
```

Comment:
It's not possible to trust in the content of our `*_settings` tables, besides the lack of data integrity, maybe in the past it was allowed to have nulled data here, which will raise an Exception unless I do coercions.

I think we should either coerce to empty strings (not null data is easier to deal with and consumes less space in the database) or allow nulls.

I decided to allow nulls, due to the existence of generic methods like the `getData('field')`, which will bring data as it was stored, perhaps with nulls.

Reply:
I'll take the same approach for all the null comments, this should be migrated to a Laravel model later, then we can rethink how things are done.

Reply:
To answer the question directly, though: there should always be an announcement title, since it's a required field for announcements. Even in an obscure case like a journal that has changed primary locales, `DataObject::getLocalizedData` will fall back on the first available piece of data if the locale precedence list doesn't provide something.